projects/packages/waf/tests/php/integration/test-waf-activation.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Activation tests.
+ *
+ * @package automattic/jetpack-waf
+ */
+
+use Automattic\Jetpack\Waf\Waf_Initializer;
+use Automattic\Jetpack\Waf\Waf_Runner;
+
+/**
+ * Activation tests.
+ */
+final class WafActivationTest extends PHPUnit\Framework\TestCase {
+
+	/**
+	 * Test setup.
+	 *
+	 * @before
+	 */
+	protected function before() {
```

Comment:
any reason to not just use `setUp()`?

Reply:
I was running into a conflict between PHPUnit wanting me to declare the return type for `setUp()`, but PHPCS not wanting the return type declared for PHP compatibility.

So it was either

```
protected function setUp() { ... }

> Fatal error: Declaration of WafActivationTest::setUp() must be compatible with PHPUnit\Framework\TestCase::setUp(): void 
> in /Users/nate/Documents/Code/jetpack/projects/packages/waf/tests/php/integration/test-waf-activation.php on line 21
```
or
```
protect function setUp(): void { ... }

> void return type is not present in PHP version 7.0 or earlier
> (PHPCompatibility.FunctionDeclarations.NewReturnTypeDeclarations.voidFound)
```

Reply:
Since you're using WorDBless master here, you should probably be using the `WorDBless\BaseTestCase` class it provides. That in turn uses [Yoast's polyfill class](https://github.com/Yoast/PHPUnit-Polyfills#option-1-yoastphpunitpolyfillstestcasestestcase) that uses `set_up()` and other snake-cased names to avoid that problem.

In your unit tests where you're still using `PHPUnit\Framework\TestCase`, I'd recommend following the same convention (although there you'll still need the `@before` directive). Or you could use Yoast's polyfill class if you want to skip the `@before` there too.

Reply:
Thanks for this info @anomiex! Updated to use the `WorDBless\BaseTestCase` and a `set_up()` method tagged with `@before` via [d53f5ca](https://github.com/Automattic/jetpack/pull/28980/commits/d53f5ca990f8ffb1a6a3008d9062b33d3b98861b).